src/shipit_pulse_listener/shipit_pulse_listener/monitoring.py

```diff
@@ -114,30 +113,35 @@ def send_report(self):
             return
 
         # Build markdown
-        total = sum([len(s) for s in self.stats.values()])
-        content = '# Pulse listener tasks for the last hour\n'
-        for group_name, tasks in self.stats.items():
-            nb_tasks = len(tasks)
-            content += GROUP_MD.format(
-                group_name,
-                100.0 * nb_tasks / total,
-                nb_tasks,
-                total,
-            )
-            content += '\n'.join([
-                TASK_MD.format(task)
-                for task in tasks
-            ])
-
-        # Send to admins
-        logger.info('Sending email to admins')
-        for email in self.emails:
-            self.notify.email({
-                'address': email,
-                'subject': 'Pulse listener tasks',
-                'content': content,
-                'template': 'fullscreen',
-            })
+        for hook_id, tasks_per_status in self.stats.items():
+            total = sum([len(tasks) for tasks in tasks_per_status.values()])
+            if len(tasks_per_status['completed']) == total:
```

Comment:
Maybe use an expression with `all()` here to increase readiblity

Reply:
I could only come up with this:
```Python
if all(len(tasks) == 0 for status, tasks in tasks_per_status.items() if status != 'completed'):
    continue
```

Which one do you prefer?

Reply:
Right, the initial one is better.